Apps/W1/EDocument/app/src/Processing/Import/PrepareDraft/PreparePurchaseEDocDraft.Codeunit.al

```diff
@@ -48,11 +48,16 @@ codeunit 6125 "Prepare Purchase E-Doc. Draft" implements IProcessStructuredData
             EDocumentPurchaseHeader."[BC] Vendor No." := Vendor."No.";
         end;
 
-        PurchaseOrder := IPurchaseOrderProvider.GetPurchaseOrder(EDocumentPurchaseHeader);
+        case Vendor."Receive E-Document To" of
```

Comment:
I would use this in the exit clause on line 62. 
exit(if Vendor.Receive E-Document To = Purchase Order ? E-Document Type::PO : E-Document Type::Invoice)

Then the rest can be left as it was. This setting should probably only influence that bit. Even if we have invoice, it would be okay to fill [BC] Purchase Order No.